staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -179,7 +184,15 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 					panicCh <- r
 				}
 			}()
-			list, err = r.listerWatcher.List(options)
+			// Attempt to gather list in chunks, if supported by listerWatcher, if not, the first
+			// list request will return the full response.
+			pager := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
+				return r.listerWatcher.List(opts)
+			}))
+			pager.PageSize = listPageSize
+			// Since listPageSize is already large, if the list expires, do not attempt a full list.
```

Comment:
I'm not convinced about this - we want finally to initialize cacher, because otherwise watch will not work at all.

Reply:
Yes, I do need to do more here. I'm concerned that a paginated list failure due to a expired error indicates that the resource count is very high since even after 5 minutes of paging 10k chunks, it still couldn't load.  But I need do some digging to understand exactly how the watch cache handles errors better. If I can't find a better way to handle this case, I'll re-enable the full list fallback.

Reply:
I've put the fallback to perform a full list back in.

Reply:
I plan to circle back on this in a future PR, potentially (a) having requests miss the watch cache until it's initial/resync list requests succeed, and/or (b) setting a max resource count on the watch cache, deactivating it for resource kinds that have more than that number of resources (e.g. 500k). If we these in place than we might be able to remove the full list fallback safely.